tools/total-contributions

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import pathlib
+import subprocess
+import sys
+from datetime import datetime
+from typing import Dict, List
+
+import requests
+
+
+def add_log(dict: Dict[str, int], input: List[str]) -> None:
+    for dataset in input:
+        name = dataset.split("\t")[1]
+        value = int(dataset.split("\t")[0])
+        if name in dict:
+            dict[name] += value
+        else:
+            dict[name] = value
+
+
+def retrieve_log(repo: str, lower_version: str, upper_version: str) -> List[str]:
+    return subprocess.check_output(
+        ["git", "shortlog", "-s", lower_version + ".." + upper_version],
+        universal_newlines=True,
+        cwd=find_path(repo),
+    ).splitlines()
+
+
+def find_version(time: str, repo: str) -> str:
+    versions_list = subprocess.check_output(
+        ["git", "tag", "-l"], universal_newlines=True, cwd=find_path(repo)
+    ).splitlines()
+    for version in versions_list:
+        version_time = subprocess.check_output(
+            ["git", "log", "-1", "--format=%ai", version],
+            universal_newlines=True,
+            cwd=find_path(repo),
+        ).split()[0]
+        if datetime.strptime(version_time, "%Y-%m-%d") >= datetime.strptime(time, "%Y-%m-%d"):
+            return version
+    return version
+
+
+def find_path(repo: str) -> str:
+    return os.path.dirname((pathlib.Path().resolve()).parents[0]) + "/" + repo
+
+
+response = requests.get(
+    "https://api.github.com/repos/zulip/zulip/releases/latest"
+)  # get latest release version
+latest_version = response.json()["tag_name"]
```

Comment:
I think this will fail when there is internet connection problem? You probably need to check for response.ok or something (I don't remember `requests`'s off the top of my head).